tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1898,18 +1889,8 @@ uint32_t DeclResultIdMapper::getTypeForPotentialAliasVar(
   if (genAlias) {
     needsLegalization = true;
 
-    const uint32_t valType = typeTranslator.translateType(type, rule);
-    // All constant/texture/structured/byte buffers are in the Uniform
-    // storage class.
-    const auto ptrType =
-        theBuilder.getPointerType(valType, spv::StorageClass::Uniform);
-
     if (info)
-      info->setStorageClass(spv::StorageClass::Uniform)
-          .setLayoutRule(rule)
-          .setValTypeId(ptrType);
-
-    return ptrType;
```

Comment:
Yep. Previously we apply the extra level of pointer in `getTypeForPotentialAliasVar`. That is good enough for **stand-alone** alias variables, but it does not support structs containing alias fields.

So I've changed to apply the extra level of pointer in `TypeTranslator` directly. So that no matter or not the alias variable is stand-alone or in a struct, the extra level of pointer will be applied appropriately and we don't need to descend recursively into a struct type in `getTypeForPotentialAliasVar` again. It simplifies stuff in a sense.

Reply:
I see. This makes more sense.